apps/s_client.c

```diff
@@ -2172,6 +2172,16 @@ int s_client_main(int argc, char **argv)
              * HTTP/d.d ddd Reason text\r\n
              */
             mbuf_len = BIO_gets(fbio, mbuf, BUFSIZZ);
+            if (mbuf_len < 12) {
+                /* '12' stands for the length of 'HTTP/d.d ddd' */
```

Comment:
Hmmm, I get your point, the compiler may optimize the strlen("string") stuff into a constant value, isn't it?